src/Components/Blazor/Blazor/src/Hosting/WebAssemblyHostBuilder.cs

```diff
@@ -102,6 +104,10 @@ private void CreateServiceProvider()
                 };
             });
 
+            // Needed for authorization
+            services.AddOptions();
+            services.TryAdd(ServiceDescriptor.Singleton(typeof(ILogger<>), typeof(WebAssemblyConsoleLogger<>)));
```

Comment:
I think you'll also have to support `ILoggerFactory`. You're going to find code in our stack that expects that, and it should be easy to add.

Reply:
I know it is fairly easy to add, but we don't have any code paths that hit it currently, and I'm trying to be minimal with this. We're always free to add more default services later.

If you have reasons to think it's going to be problematic if we don't, let me know, otherwise I'll leave this as-is.

Reply:
I can't really articulate a specific scenario. This just feels lazy - because we know what we would do if people complain. We could prevent them from having to log and issue and wait for us to fix it by just doing a little thing now.

Reply:
Yeah, we want to have a similar set of services in both environments, otherwise when I try to use a component that does logging, then I end up having to register it myself, making the experience different in the two environments.

Reply:
OK, I've added a super-minimal `ILoggerFactory`.

> Yeah, we want to have a similar set of services in both environments

I'd be cautious about thinking about it like that. The environments have very different capabilities, goals, and characteristics. Most of the server-side services are for server-side application-server concerns, so a large proportion of them won't be present on WebAssembly by design.